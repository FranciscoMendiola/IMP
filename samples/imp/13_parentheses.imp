x := (a + b) * (c - d);
if (x = 0) then
  skip
else
  y := (x + 1) * 2
