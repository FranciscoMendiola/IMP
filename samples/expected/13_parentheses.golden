Token {tokenKind = Id, lexeme = "x", position = 0}
Token {tokenKind = Assign, lexeme = ":=", position = 2}
Token {tokenKind = LPar, lexeme = "(", position = 5}
Token {tokenKind = Id, lexeme = "a", position = 6}
Token {tokenKind = Plus, lexeme = "+", position = 8}
Token {tokenKind = Id, lexeme = "b", position = 10}
Token {tokenKind = RPar, lexeme = ")", position = 11}
Token {tokenKind = Times, lexeme = "*", position = 13}
Token {tokenKind = LPar, lexeme = "(", position = 15}
Token {tokenKind = Id, lexeme = "c", position = 16}
Token {tokenKind = Minus, lexeme = "-", position = 18}
Token {tokenKind = Id, lexeme = "d", position = 20}
Token {tokenKind = RPar, lexeme = ")", position = 21}
Token {tokenKind = Semi, lexeme = ";", position = 22}
Token {tokenKind = KIf, lexeme = "if", position = 24}
Token {tokenKind = LPar, lexeme = "(", position = 27}
Token {tokenKind = Id, lexeme = "x", position = 28}
Token {tokenKind = Eq, lexeme = "=", position = 30}
Token {tokenKind = IntLit, lexeme = "0", position = 32}
Token {tokenKind = RPar, lexeme = ")", position = 33}
Token {tokenKind = KThen, lexeme = "then", position = 35}
Token {tokenKind = KSkip, lexeme = "skip", position = 42}
Token {tokenKind = KElse, lexeme = "else", position = 47}
Token {tokenKind = Id, lexeme = "y", position = 54}
Token {tokenKind = Assign, lexeme = ":=", position = 56}
Token {tokenKind = LPar, lexeme = "(", position = 59}
Token {tokenKind = Id, lexeme = "x", position = 60}
Token {tokenKind = Plus, lexeme = "+", position = 62}
Token {tokenKind = IntLit, lexeme = "1", position = 64}
Token {tokenKind = RPar, lexeme = ")", position = 65}
Token {tokenKind = Times, lexeme = "*", position = 67}
Token {tokenKind = IntLit, lexeme = "2", position = 69}
